packages/instantsearch.js/src/connectors/refinement-list/connectRefinementList.ts

```diff
@@ -562,4 +559,27 @@ As this is not supported, please make sure to only use this attribute with one o
     };
   };
 
+function removeEmptyRefinementsFromUiState(indexUiState: IndexUiState) {
```

Comment:
does removing this function break any tests?

Reply:
Yes, it adds empty values in widget's ui states, which are not expected by tests, and can in some instance show with the default router. I've added this method in each connector for now, but I'd like to see if I can clean these empty refinements from a single location instead.

Reply:
maybe it can be done in the index widget after getUiState has been called, so you only need to iterate once, instead of for every widget (although getWidgetUiState is called in multiple places of course)

Reply:
It wasn't feasible to remove empty attributes at a later point in the index widget, as:
- some parts of the call directly use results from connectors' `getWidgetUiState()`
- the predicates used to determine wether attributes are considered empty differs between connectors